### PR TITLE
(feature) Add React router to the frontend

### LINKED ISSUE #37: Set up React Router

### PR DESCRIPTION
**Pull request by azangru**
- added React router
- added two routes: the root route ('/') and the sprint route ('/sprint')
- added a dummy view for the root route and a link that leads to the
  sprint view
- set the Router to use HTML5 history API (to avoid '#'-signs in urls)
- in order to do that, added the index.ejs template view on the server
  side and instructed the server to render this view in response to all
  get requests
- (additionally) Added .nvmrc to the root folder in order for nvm to
  automatically use the correct version of node

THIS PR MAKES OBSOLETE:
- index.html in `/frontend`
- gulp task `frontend:build:html`
  but I haven't yet removed them

Closes #37

### CODE REVIEW

File: frontend/src/styles.css [hunk 1]
```diff
@@ -6,4 +6,8 @@ html, body{
   height: 100%;
 }
 
+.appContainer{
```
Comment (azangru) [inline frontend/src/styles.css:9]: Currently I have a div with id `app` in html, and am rendering React components into it.

This class (with a confusing name) is for another div — the main React div (which will be rendered in request of root route, and within which all other React components will be rendered).

I guess, I can just remove the `div id='app'` from index.ejs and render React components directly into the body.

File: frontend/server/server.js [hunk 2]
```diff
@@ -1,11 +1,22 @@
...
 
 let server = http.createServer(app);
 
+app.get('*', (req, res) => {
+  res.render('index.ejs');
+});
+
+// app.get('*', (req, res) => {
+//   res.sendFile(path.join(__dirname, '../dist/index.html'));
+// });
```
Comment (azangru) [inline frontend/server/server.js:18]: Tested this method (commented out above) as an alternative to res.render('index.ejs'). It works, but unless you have particular objections, I would prefer rendering the template on the server.

File: frontend/src/styles.css (hunk 1 above)
Comment (wezleytsai) [inline frontend/src/styles.css:9]: That makes sense - that's also how it's done in the `react-router` readme example

File: frontend/src/routes.jsx.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react/addons';
+import Router from 'react-router';
+import App from './components/app.jsx.js';
+import SprintBoard from './components/sprintboard/board.jsx.js';
+
+let Route = Router.Route;
+// -- WE MAY NEED THIS AT SOME LATER POINT FOR A 404 page:
+// let NotFoundRoute = Router.NotFoundRoute;
+
+let routes = (
```
Comment (wezleytsai) [inline frontend/src/routes.jsx.js:12]: Question: how is the default route determined? Is it your `<Link to="sprint">Sprint</Link>` code in app.jsx.js?

File: frontend/src/components/app.jsx.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React from 'react/addons';
+import {RouteHandler} from 'react-router';
+import {Link} from 'react-router';
+
+let App = React.createClass({
+  render: () => {
+    return (
+      <div className="appContainer">
+        <RouteHandler/>
+        <Link to='sprint'>Sprint</Link>
```
Comment (wezleytsai) [inline frontend/src/components/app.jsx.js]: OCD: are we supposed to be using double quotes within all HTML tags - you have double quotes in routes.jsx.js

**Pull request closed without merging**